Increase timeout to 600s when using --cicd (fixes #65)

CarlosFelipeOR opened the pull request:
This PR adds a `--cicd` flag that increases the default bash timeout from  300s to 600s when running tests in CI/CD environments. This addresses  timeout issues with long-running operations like OPTIMIZE TABLE that can  take more than 5 minutes when the full test suite runs in parallel.

Changes:
- Add `--cicd` argument to argparser
- Modify `Cluster.bash()` to use 600s timeout when `cicd=True`
- Add `--cicd` flag to GitHub Actions workflows via reusable-suite.yml
- Update `Cluster.__init__()` and `create_cluster()` to accept `cicd` parameter
- Update `CaptureClusterArgs` to pass `cicd` to cluster creation

Fixes #65.

vzakaznikov commented on helpers/cluster.py:
```diff
@@ -1784,11 +1786,14 @@ def shell(self, node, timeout=300):
         to a specific node.
         :param node: name of the service
         """
+        if timeout is None:
```
Or maybe just call it simple ```NONE``` without the underscores.

CarlosFelipeOR commented:
All requested changes have been applied:

- Introduced a unique `object()` sentinel (`NONE`) for the `timeout` argument to clearly distinguish between an omitted value and an explicit `None`.
- Updated the default timeout logic to rely on the sentinel instead of `None`.
- Adjusted the `--cicd` help message by removing the fixed "Timeout=600s" note, since the flag may cover additional behaviors in the future.

@vzakaznikov, please take a look.
